### ISSUE TITLE
TypeError: Assignment to constant variable when using native solc

### ISSUE THREAD

**Issue by soylent** [trufflesuite/truffle#2817]
## Issue

`truffle test` fails when using native solc

## Steps to Reproduce

1. `truffle unbox metacoin`
2. add the following to `truffle-config.js`:
```js
compilers: {
  solc: {
    version: "native"
  }
}
```
3. `truffle test`

## Actual Results

Solidity tests fail:
```
Compiling your contracts...
===========================
> Compiling ./contracts/ConvertLib.sol
> Compiling ./contracts/MetaCoin.sol
> Compiling ./contracts/Migrations.sol
> Compiling ./test/TestMetaCoin.sol
> Artifacts written to /tmp/test-202018-12973-k1zmz.ktj20b
> Compiled successfully using:
   - solc: 0.6.2+commit.bacdbe57.Linux.g++

...

 1) TestMetaCoin
       "before all" hook: prepare suite:
     TypeError: Assignment to constant variable.
      at Context.compileNewAbstractInterface (/home/user/.config/yarn/global/node_modules/truffle/build/webpack:/packages/core/lib/testing/soliditytest.js:112:1)
      at Context.<anonymous> (/home/user/.config/yarn/global/node_modules/truffle/build/webpack:/packages/core/lib/testing/soliditytest.js:23:1)
      at process._tickCallback (internal/process/next_tick.js:68:7)
```

## Environment

* Operating System: Arch
* solc: 0.6.2+commit.bacdbe57.Linux.g++
* Truffle v5.1.11 (core: 5.1.11)
* Node v10.16.0
* Web3.js v1.2.1

**Comment by CruzMolina** [comment]
Hey @soylent ! This should be resolved in v5.1.12 via #2816.

Please let us know if this is still an issue in v5.1.12.

**Comment by soylent** [comment]
Thanks, @CruzMolina The issue is fixed in the new version.
